### PR TITLE
[aws refactor] aws-sdk-cpp: Review dependencies, add new version and remove old one

### PR DESCRIPTION
**Pull request by danimtb**
Related to https://github.com/conan-io/conan-center-index/issues/24831

### CODE REVIEW

**Comment by uilianries** [comment]
I just tried to build this PR for Android and it failed in the configure:
[aws-sdk-cpp-1.11.352-android.log](https://github.com/user-attachments/files/16688066/aws-sdk-cpp-1.11.352-android.log)

However, is better than in master branch. When trying to build the master branch, not even `aws-c-common/0.6.11` does not build.

File: recipes/aws-sdk-cpp/all/conanfile.py [hunk 1]
```diff
@@ -336,15 +336,21 @@ def layout(self):
...
             self.requires("aws-c-http/0.8.1")
             self.requires("aws-c-io/0.14.7")
-            self.requires("aws-crt-cpp/0.26.9", transitive_headers=True)
+            self.requires("aws-c-mqtt/0.10.3")
             if self.options.get_safe("s3-crt"):
                 self.requires("aws-c-s3/0.5.5")
+            # self.requires("aws-c-sdkutils/0.1.15")
+            self.requires("aws-checksums/0.1.18")
+            # missing aws-lc
+            # self.requires("s2n/1.4.16")
```
Comment (danimtb) [inline recipes/aws-sdk-cpp/all/conanfile.py:353]: why not enable aws-c-sdkutils? I see it is listed here https://github.com/aws/aws-sdk-cpp/blob/1b9f1f48db4bb34967508afcbb91b1a1be992b3c/prefetch_crt_dependency.sh#L17

Also s2n https://github.com/aws/aws-sdk-cpp/blob/1b9f1f48db4bb34967508afcbb91b1a1be992b3c/prefetch_crt_dependency.sh#L20

**Review by danimtb — COMMENTED** [review]
Let's see if it builds for all configurations. Meanwhile, a couple of questions regarding the disabled requirements. Thanks!

File: recipes/aws-sdk-cpp/all/conanfile.py [hunk 2]
```diff
@@ -540,7 +559,7 @@ def package_info(self):
         else:
             self.cpp_info.components["core"].requires.extend(["libcurl::curl", "openssl::openssl"])
 
-        if self.settings.os in ["Linux", "FreeBSD"]:
+        if self.settings.os in ["Linux", "FreeBSD", "Android"]:
             self.cpp_info.components["core"].system_libs.append("atomic")
```
Comment (cynix) [inline recipes/aws-sdk-cpp/all/conanfile.py:563]: Is it possible to remove the `libatomic` dependency for architectures that don’t need it?

**Comment by DevAGCO** [comment]
Hey guys,
Thank you for the work you put in this. We are so much looking forward to it! :)

**Comment by uilianries** [comment]
Hello again! 

I just tried to build again for Android, using the commit [90e7f69](https://github.com/conan-io/conan-center-index/pull/24924/commits/90e7f695b48f4f8c29c9035cee84d5494dad319a). It's no longer possible this, because some dependencies are only supported on Linux:

```
======== Exporting recipe to the cache ========
aws-sdk-cpp/1.11.352: Exporting package recipe: /home/uilian/Development/conan/conan-center-index/recipes/aws-sdk-cpp/all/conanfile.py
aws-sdk-cpp/1.11.352: exports: File 'conandata.yml' found. Exporting it...
aws-sdk-cpp/1.11.352: Calling export_sources()
aws-sdk-cpp/1.11.352: Copied 1 '.py' file: conanfile.py
aws-sdk-cpp/1.11.352: Copied 1 '.yml' file: conandata.yml
aws-sdk-cpp/1.11.352: Copied 3 '.patch' files: 1.11.352-0001-pulseaudio.patch, 1.11.352-0002-disable-sort-links.patch, 1.9.100-0002-aws-plugin-conf.patch
aws-sdk-cpp/1.11.352: Exported to cache folder: /home/uilian/.conan2/p/aws-sa2bddb21d2c56/e
aws-sdk-cpp/1.11.352: Exported: aws-sdk-cpp/1.11.352#c4dcce5cadffbdaddb13c0612e5f3328 (2024-09-05 09:37:06 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.cppstd=17
compiler.libcxx=c++_static
compiler.version=17
os=Android
os.api_level=32
[tool_requires]
*: cmake/3.29.7, android-ndk/r26d
[conf]
tools.build:verbosity=verbose
tools.cmake.cmaketoolchain:generator=Ninja
tools.compilation:verbosity=verbose
tools.system.package_manager:mode=install
tools.system.package_manager:sudo=yes

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
[tool_requires]
*: cmake/3.29.7
[conf]
tools.build:verbosity=verbose
tools.cmake.cmaketoolchain:generator=Ninja
tools.compilation:verbosity=verbose
tools.system.package_manager:mode=install
tools.system.package_manager:sudo=yes


======== Computing dependency graph ========
Graph root
    cli
Requirements
    autoconf/2.71#f9307992909d7fb3df459340f1932809 - Cache
    automake/1.16.5#058bda3e21c36c9aa8425daf3c1faf50 - Cache
    aws-c-auth/0.7.16#d1c779b29e87d9b9e6564a1b4a5a2e28 - Cache
    aws-c-cal/0.6.14#3ed4cc4d6b5ebe63bcbed6fd0fa9951f - Cache
    aws-c-common/0.9.15#c44a2ce5f1e858caf1a5b692078755eb - Cache
    aws-c-compression/0.2.18#c86b8f00cee8d572cd71835e9842405f - Cache
    aws-c-event-stream/0.4.2#a390c1c504451a7eeab447bfbb4a9e47 - Cache
    aws-c-http/0.8.1#2dbdd363d5daf73ead6877310e406ebe - Cache
    aws-c-io/0.14.7#fd5eacb9a36c7d2277ebcdb8972dc7c6 - Cache
    aws-c-mqtt/0.10.3#a4d43fcbcc1e0a91079ec93691dd64d2 - Cache
    aws-c-s3/0.5.5#cae5e2180e586fe5c10a278999a7ca8b - Cache
    aws-c-sdkutils/0.1.15#6cfaa1f582765f5fabf9609f7626522c - Cache
    aws-checksums/0.1.18#7039b79391bd012855848816cb663464 - Cache
    aws-crt-cpp/0.26.9#a63cd7ec7a0b15cfa417cecc4a9cf6e9 - Cache
    aws-sdk-cpp/1.11.352#c4dcce5cadffbdaddb13c0612e5f3328 - Cache
    flac/1.4.2#d4051f8b37ee93725dea40dfd6a10168 - Cache
    libalsa/1.2.10#e64d5e1ced869a2f676145bab4f4a181 - Cache
    libcap/2.69#7ef2d60864c2e58e89db957be936dc49 - Cache
    libcurl/8.9.1#9e940595377416e8d16e813a21e687a0 - Cache
    libiconv/1.17#73fefc1b696e069df90fd1d18aa63edd - Cache
    libmp3lame/3.100#44b12d19316eb2b223d98d3e75dae438 - Cache
    libsndfile/1.2.2#b3662f832e29507e8d9840a43737cc14 - Cache
    libtool/2.4.7#08316dad5c72c541ed21e039e4cf217b - Cache
    m4/1.4.19#b38ced39a01e31fef5435bc634461fd2 - Cache
    mpg123/1.31.2#ce831c936b2284e2066ab3dc58a2628e - Cache
    ogg/1.3.5#062626875f5c8c59f069f76f148098ef - Cache
    openssl/3.3.1#279e86dc2a7dded8c1fe8512e775bdb1 - Cache
    opus/1.4#54631f551fc450783fb2df8cd63f80a2 - Cache
    pulseaudio/14.2#30afeff38a7770cb9cda17a66da49a19 - Cache
    s2n/1.4.1#4b875ac0b03e47ba9f5d673f0c4298d1 - Cache
    vorbis/1.3.7#37e58f52e59a6232199b34ef402714a6 - Cache
    xorg/system#f626cbdd0ba57d9c53bce2d8b9362fad - Cache
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2 - Cache
Build requirements
    android-ndk/r26d#0a8a020a2e0ce2edbda6ff226a33a23a - Cache
    autoconf/2.71#f9307992909d7fb3df459340f1932809 - Cache
    automake/1.16.5#058bda3e21c36c9aa8425daf3c1faf50 - Cache
    cmake/3.29.7#cbf5446442af81d8f7e1b6c719e5982b - Cache
    gettext/0.21#a2bd3513f212013764c9040f4c70ed69 - Cache
    gnu-config/cci.20210814#dc430d754f465e8c74463019672fb97b - Cache
    libiconv/1.17#73fefc1b696e069df90fd1d18aa63edd - Cache
    libtool/2.4.7#08316dad5c72c541ed21e039e4cf217b - Cache
    m4/1.4.19#b38ced39a01e31fef5435bc634461fd2 - Cache
    meson/1.2.2#29cf77c283f6af08818f6fd8068c37e8 - Cache
    ninja/1.12.1#fd583651bf0c6a901943495d49878803 - Cache
    pkgconf/2.0.3#f996677e96e61e6552d85e83756c328b - Cache
    pkgconf/2.1.0#27f44583701117b571307cf5b5fe5605 - Cache
Resolved version ranges
    libcurl/[>=7.78.0 <9]: libcurl/8.9.1
    ninja/[>=1.10.2 <2]: ninja/1.12.1
    openssl/[>=1.1 <4]: openssl/3.3.1
    zlib/[>=1.2.11 <2]: zlib/1.3.1

======== Computing necessary packages ========
flac/1.4.2: Main binary package 'aac4019c4f5babf056427efb8e6e59bca90b5b1b' missing
flac/1.4.2: Checking 11 compatible configurations
flac/1.4.2: Compatible configurations not found in cache, checking servers
flac/1.4.2: 'd19dc82862c927a665d5b7c4c6eca5613273eb66': compiler.cppstd=98
flac/1.4.2: '6cc74ee5bddc94c9c7338beac8f29e719b6919ae': compiler.cppstd=gnu98
flac/1.4.2: 'c724c078618a3338124e1687a52f2e6f4fbbaae9': compiler.cppstd=11
flac/1.4.2: '159708172fa21512153bf52903234f56bc6156af': compiler.cppstd=gnu11
flac/1.4.2: '49e48959384d10e0ff858c34e0e31bb9747dbd98': compiler.cppstd=14
flac/1.4.2: '0e809fbffff9b0c996cb14c48d555db9bed5b2a3': compiler.cppstd=gnu14
flac/1.4.2: '5d16bef129ec8be417a89c455cb8302af8df3201': compiler.cppstd=gnu17
flac/1.4.2: '1a610d4fbcb6291c4f16ce1ea1c4f3c2723c1dd2': compiler.cppstd=20
flac/1.4.2: '345efc13d25e18c4754b0184070cd48891c0fe6a': compiler.cppstd=gnu20
flac/1.4.2: '0d7464a5aee1cf1c377176f02e7c3d5a98fe9636': compiler.cppstd=23
flac/1.4.2: 'cbaf324fe44364ad82ac9b770ee7e8bc69cca925': compiler.cppstd=gnu23
libsndfile/1.2.2: Main binary package '55636d17b6990289bfbedaec87a3865a94a0add4' missing
libsndfile/1.2.2: Checking 11 compatible configurations
libsndfile/1.2.2: Compatible configurations not found in cache, checking servers
libsndfile/1.2.2: '16eee1caff7c82e648eea70230a912a16ae1b99f': compiler.cppstd=98
libsndfile/1.2.2: '493cfc67b4d9ac7662e974b9040326203275b3cb': compiler.cppstd=gnu98
libsndfile/1.2.2: '988a1c97e055f484625ca740ed658a229ff55444': compiler.cppstd=11
libsndfile/1.2.2: '10926c695547b21a033fa41f1dc7c465448d47bb': compiler.cppstd=gnu11
libsndfile/1.2.2: 'b8a20f75776983abcce65682ce5bb367f11da34c': compiler.cppstd=14
libsndfile/1.2.2: '0be9810e58beed22874fa7c9c51c9785c26a1e4c': compiler.cppstd=gnu14
libsndfile/1.2.2: '90d11e391e6f34bcd3fef215e4a5ce6e5e5c5cc1': compiler.cppstd=gnu17
libsndfile/1.2.2: 'e5b6afc73ac56f76e2a2bb4c9468828117b5dd7d': compiler.cppstd=20
libsndfile/1.2.2: '9b64fbfc0793a86ad5e7e88efa24db328f9159e9': compiler.cppstd=gnu20
libsndfile/1.2.2: 'ecae140538d0b97b68a9b47084c282bcf512ee26': compiler.cppstd=23
libsndfile/1.2.2: 'b684e6c1ad3a1175f0a325dada242bc004962081': compiler.cppstd=gnu23
aws-sdk-cpp/1.11.352: Forced build from source
Requirements
    autoconf/2.71#f9307992909d7fb3df459340f1932809:da39a3ee5e6b4b0d3255bfef95601890afd80709#5b77f70c17ad1741f5845d4e468a347e - Cache
    automake/1.16.5#058bda3e21c36c9aa8425daf3c1faf50:e176f8417bf0357eeab5ecede3608cab6e8d3286 - Missing
    aws-c-auth/0.7.16#d1c779b29e87d9b9e6564a1b4a5a2e28:7af801ebb71fa8aab5e774adc256260defa29763#b13ba6193e135336b22ee0c62599ee7d - Cache
    aws-c-cal/0.6.14#3ed4cc4d6b5ebe63bcbed6fd0fa9951f:a353b328f3eb9f70b6ea49237c1688b4f3369537#4d288240067fcafbf184f2e3edf9e3e9 - Cache
    aws-c-common/0.9.15#c44a2ce5f1e858caf1a5b692078755eb:a1f50cef425cf4d50314539190b53ae3557ee5b1#bf118c090ac0e2e8a7d910d046ae2a0a - Cache
    aws-c-compression/0.2.18#c86b8f00cee8d572cd71835e9842405f:46cc8bf67bf66b3d9004386dc39fe69a953a6d64#6749c3d262dd30a54828bdac4be61536 - Cache
    aws-c-event-stream/0.4.2#a390c1c504451a7eeab447bfbb4a9e47:bbce6e7108210bed237a9f5e48a282b4aa26f006#de54a2204b83f44c00df043f1d67fa90 - Cache
    aws-c-http/0.8.1#2dbdd363d5daf73ead6877310e406ebe:84fd6221ddafe98131b193b0bb01ce5bc3f828b0#7ec40175968a0845db375ec74e455850 - Cache
    aws-c-io/0.14.7#fd5eacb9a36c7d2277ebcdb8972dc7c6:fa869cef2754b66ad70ff831d417c0557dcd90b0#2b0520e5ee1d45d18359c5d01ccc9776 - Cache
    aws-c-mqtt/0.10.3#a4d43fcbcc1e0a91079ec93691dd64d2:da29f1ad058fbcd10c1ce3dd9d0a63c63cb930db#16aa865cca07d05caab6ee6f2082e5d7 - Cache
    aws-c-s3/0.5.5#cae5e2180e586fe5c10a278999a7ca8b:2d8cbffda35dc9806befd40656be95157c5fbaef#91e373c2e1659ee4dba9168cbd699d97 - Cache
    aws-c-sdkutils/0.1.15#6cfaa1f582765f5fabf9609f7626522c:46cc8bf67bf66b3d9004386dc39fe69a953a6d64#42053eb757c7295de81d60d9dc08815e - Cache
    aws-checksums/0.1.18#7039b79391bd012855848816cb663464:46cc8bf67bf66b3d9004386dc39fe69a953a6d64#6b9a29534c50f6c7cb0c8fd24d79372b - Cache
    aws-crt-cpp/0.26.9#a63cd7ec7a0b15cfa417cecc4a9cf6e9:6357a9561c5c806c90f5b63c27b431a8636abc9f#823be417d699ca214fa26232b48f3bc7 - Cache
    aws-sdk-cpp/1.11.352#c4dcce5cadffbdaddb13c0612e5f3328:85ad8c4163dbb1cd213d0aa976b802266be66a78 - Build
    flac/1.4.2#d4051f8b37ee93725dea40dfd6a10168:aac4019c4f5babf056427efb8e6e59bca90b5b1b - Missing
    libalsa/1.2.10#e64d5e1ced869a2f676145bab4f4a181:8abf0d122a86079cee9792d621fe473a152184f4 - Invalid
    libcap/2.69#7ef2d60864c2e58e89db957be936dc49:2845940b18c766a425229114a584d50898a0e076 - Invalid
    libcurl/8.9.1#9e940595377416e8d16e813a21e687a0:544aaca6ad686a9e1d0deba59cf340905296cc7f#4d55cd2faea7311406f74cdd8b6601ac - Cache
    libiconv/1.17#73fefc1b696e069df90fd1d18aa63edd:1fb490abea0c07fe60f9417ffad5cf9f74dde6ec#6c7b5655403d2864bdb3741b8cf7e4ec - Cache
    libmp3lame/3.100#44b12d19316eb2b223d98d3e75dae438:1fb490abea0c07fe60f9417ffad5cf9f74dde6ec#a686ebb0321e975bd71e6a0cdaf9f21f - Cache
    libsndfile/1.2.2#b3662f832e29507e8d9840a43737cc14:55636d17b6990289bfbedaec87a3865a94a0add4 - Missing
    libtool/2.4.7#08316dad5c72c541ed21e039e4cf217b:1fb490abea0c07fe60f9417ffad5cf9f74dde6ec - Missing
    m4/1.4.19#b38ced39a01e31fef5435bc634461fd2:b1f00bf3a529e316c1a53a32674c14663a42d79c - Missing
    mpg123/1.31.2#ce831c936b2284e2066ab3dc58a2628e:e0e027d40f9a759c7d1d50c92698cc48cbdb95ce - Missing
    ogg/1.3.5#062626875f5c8c59f069f76f148098ef:1fb490abea0c07fe60f9417ffad5cf9f74dde6ec#10561eb4f5c9a7ac210f46f40300f421 - Cache
    openssl/3.3.1#279e86dc2a7dded8c1fe8512e775bdb1:1dc48a97931ebb534503cefbc936d61ec31448d0#bcf8b85ace883b30260ba2a17e84ef0f - Cache
    opus/1.4#54631f551fc450783fb2df8cd63f80a2:bc89a0ba160db5d1f8cdb885d57c03e577055ee5#117ea8335c69b15ca87eff8130e2b3aa - Cache
    pulseaudio/14.2#30afeff38a7770cb9cda17a66da49a19:0c649ec101332c2f52a32d94dda829cfd4e8cbfb - Invalid
    s2n/1.4.1#4b875ac0b03e47ba9f5d673f0c4298d1:63945c9d4dcdfebf1910321b1b5095a58c63991d#a6d88d40c8f69e857769e650c5c39ace - Cache
    vorbis/1.3.7#37e58f52e59a6232199b34ef402714a6:f252bd52f5fda8b42b319d30d6af264b5691a5b4#e6ba3ae734f0900033353454f58cf899 - Cache
    xorg/system#f626cbdd0ba57d9c53bce2d8b9362fad:da39a3ee5e6b4b0d3255bfef95601890afd80709 - Invalid
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2:1fb490abea0c07fe60f9417ffad5cf9f74dde6ec#c166df81ec333aac702db77c5d0b9829 - Cache
Build requirements
    android-ndk/r26d#0a8a020a2e0ce2edbda6ff226a33a23a:63fead0844576fc02943e16909f08fcdddd6f44b#b049c823c3ef83c1c734cd5760bd11aa - Cache
    cmake/3.29.7#cbf5446442af81d8f7e1b6c719e5982b:63fead0844576fc02943e16909f08fcdddd6f44b#a394ed2616f3b9d4c653c510e5838734 - Cache
Skipped binaries
    autoconf/2.71, automake/1.16.5, gettext/0.21, gnu-config/cci.20210814, libiconv/1.17, libtool/2.4.7, m4/1.4.19, meson/1.2.2, ninja/1.12.1, pkgconf/2.0.3, pkgconf/2.1.0
ERROR: There are invalid packages:
libcap/2.69: Invalid: libcap only supports Linux
xorg/system: Invalid: This recipe supports only Linux and FreeBSD
libalsa/1.2.10: Invalid: libalsa/1.2.10 only supports Linux
pulseaudio/14.2: Invalid: pulseaudio supports only linux currently
```

I would suggest adding a filter in the recipe, by removing those dependencies when building for Android.

File: recipes/aws-sdk-cpp/all/conanfile.py [hunk 3]
```diff
@@ -358,9 +364,19 @@ def requirements(self):
         if self.settings.os != "Windows":
             self.requires("openssl/[>=1.1 <4]")
             self.requires("libcurl/[>=7.78.0 <9]")
-        if self.settings.os in ["Linux", "FreeBSD"]:
+        if self.settings.os in ["Linux", "FreeBSD", "Android"]:
```
Comment (uilianries) [inline recipes/aws-sdk-cpp/all/conanfile.py:367]:
```suggestion
        if self.settings.os in ["Linux", "FreeBSD"]:
```

Pulseaudio has transitive dependencies no supported for Android.

File: recipes/aws-sdk-cpp/all/conanfile.py (hunk 2 above)
Comment (AbrilRBS) [reply recipes/aws-sdk-cpp/all/conanfile.py]: yes, sorry about that!

**Comment by uilianries** [comment]
Doing a new build for Android, using the commit [d046db4](https://github.com/conan-io/conan-center-index/pull/24924/commits/d046db47ab7a52e06b0e253ec9a4114d5b2feb03):

It's working now to the version 1.11.352: [aws-sdk-cpp-1.11.352-android.log](https://github.com/user-attachments/files/16892910/aws-sdk-cpp-1.11.352-android.log)

Still need to replicate the same patch change to `1.8.130-0004-improve-pulseaudio-detection.patch`

Just tested on Linux too, and the logic is correct, I can find `-DPULSE` when building with GCC11:

```
[76/104] /usr/bin/c++ -DAWS_SDK_VERSION_MAJOR=1 -DAWS_SDK_VERSION_MINOR=11 -DAWS_SDK_VERSION_PATCH=352 -DAWS_TEST_REGION=US_EAST_1 -DENABLED_REQUEST_COMPRESSION -DENABLED_ZLIB_REQUEST_COMPRESSION -DENABLE_CURL_CLIENT -DENFORCE_TLS_V1_2 -DFLAC__NO_DLL -DHAS_PATHCONF -DHAS_UMASK -DPLATFORM_LINUX -DPULSE -D_REENTRANT -I/home/uilian/.conan2/p/b/aws-sa94a2038fce6c/b/src/src/aws-cpp-sdk-text-to-speech/include -I/home/uilian/.conan2/p/b/aws-sa94a2038fce6c/b/src/generated/src/aws-cpp-sdk-polly/include -I/home/uilian/.conan2/p/b/aws-sa94a2038fce6c/b/src/src/aws-cpp-sdk-core/include -I/home/uilian/.conan2/p/libcu76c3f898a68f0/p/include -isystem /home/uilian/.conan2/p/aws-c991b643411b88/p/include -isystem /home/uilian/.conan2/p/aws-c9f3f5aaa0ea52/p/include -isystem /home/uilian/.conan2/p/aws-cd28770ab3e858/p/include -isystem /home/uilian/.conan2/p/aws-c3d52e11cce664/p/include -isystem /home/uilian/.conan2/p/aws-cc427c62929a99/p/include -isystem /home/uilian/.conan2/p/aws-cf2a758785e82e/p/include -isystem /home/uilian/.conan2/p/aws-cf6764943db61c/p/include -isystem /home/uilian/.conan2/p/aws-cd4bd7597d633d/p/include -isystem /home/uilian/.conan2/p/opens47f8922e04a22/p/include -isystem /home/uilian/.conan2/p/b/zlib71710c3ddfc5c/p/include -isystem /home/uilian/.conan2/p/aws-cd55b54da5d139/p/include -isystem /home/uilian/.conan2/p/aws-c2a4b3980ad616/p/include -isystem /home/uilian/.conan2/p/aws-c3c06b6ea45294/p/include -isystem /home/uilian/.conan2/p/pulse1c227a25123e8/p/include -m64 -O3 -DNDEBUG -std=gnu++17 -fPIC -fno-exceptions -std=c++11 -fPIC -Wall -pedantic -Wextra -Werror -MD -MT src/aws-cpp-sdk-text-to-speech/CMakeFiles/aws-cpp-sdk-text-to-speech.dir/source/text-to-speech/TextToSpeechManager.cpp.o -MF src/aws-cpp-sdk-text-to-speech/CMakeFiles/aws-cpp-sdk-text-to-speech.dir/source/text-to-speech/TextToSpeechManager.cpp.o.d -o src/aws-cpp-sdk-text-to-speech/CMakeFiles/aws-cpp-sdk-text-to-speech.dir/source/text-to-speech/TextToSpeechManager.cpp.o -c /home/uilian/.conan2/p/b/aws-sa94a2038fce6c/b/src/src/aws-cpp-sdk-text-to-speech/source/text-to-speech/TextToSpeechManager.cpp
```

**Comment by AbrilRBS** [comment]
> Thank you for the work you put in this. We are so much looking forward to it! :)

Thanks for the kind words @DevAGCO!

Could you test the new version here on your end, and tell us if it meets your usage expectations after some testing? I think it could be very valuable in this case, as we've split adding some of the new sdks etc for a follow-up PR

**Comment by DevAGCO** [comment]
> > Thank you for the work you put in this. We are so much looking forward to it! :)
> 
> Thanks for the kind words @DevAGCO!
> 
> Could you test (even a minor one would be ace!) the new version here on your end, and tell us if it meets your usage expectations after some testing? I think it could be very valuable in this case, as we've split adding some of the new sdks etc for a follow-up PR

I'd love to help! We are using this package with cmake, cross-compiling with MingW. We use the syntax:
`cmake ... -DCMAKE_PROJECT_TOP_LEVEL_INCLUDES=conan_provider.cmake`
so I'm not sure how to build your branch in that setup.

**Review by AbrilRBS — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM.

**Comment by danimtb** [comment]
@maksim-petukhov we are looking at it, thanks for your quick report.

**Comment by AbrilRBS** [comment]
@maksim-petukhov thanks a lot for the heads up, sorry for the confussion, you're completely right, it seems my local testing was lacking. I've proposed https://github.com/conan-io/conan-center-index/pull/25401 to fix the issue

**Pull request merged**